parser.go

```diff
@@ -2335,7 +2348,11 @@ func mustGetMetricName(lfss []*labelFilterExpr) string {
 	}
 	lfs := lfss[0]
 	if lfs.Label != "__name__" || lfs.Value == nil || len(lfs.Value.tokens) != 1 {
-		return ""
+		if lfs.IsPossibleMetricName {
+			return lfs.Label
+		} else {
+			return ""
+		}
```

Comment:
```suggestion
		if lfs.IsPossibleMetricName {
			return lfs.Label
		} 
		return ""
```

reducing indentations also reduces cognitive load when reading the code.

Reply:
This is an interesting take.  I was of the mindset to make the else behavior explicit.  I'll change to follow our convention especially since it will reduce cognitive load.